prog/target.go

```diff
@@ -305,13 +305,13 @@ func (pg *Builder) Append(c *Call) error {
 	return nil
 }
 
-func (pg *Builder) Allocate(size uint64) uint64 {
-	return pg.ma.alloc(nil, size)
+func (pg *Builder) Allocate(size uint64, alignment uint64) uint64 {
+	return pg.ma.alloc(nil, size, alignment)
 }
 
 func (pg *Builder) AllocateVMA(npages uint64) uint64 {
 	psize := pg.target.PageSize
-	addr := pg.ma.alloc(nil, (npages+1)*psize)
+	addr := pg.ma.alloc(nil, (npages+1)*psize, 1)
```

Comment:
What we are trying to do here is exactly providing alignment. It now makes sense to request PageSize alignment directly.

Reply:
So something like
`return pg.ma.alloc(nil, npages * pg.target.PageSize, pg.target.PageSize)`
makes more sense now, right? Updated the commit to reflect this.